x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/page_content.tsx

```diff
@@ -80,6 +88,7 @@ export function StreamDetailEnrichmentContentImpl() {
   const context = useKibana();
   const { appParams, core } = context;
 
+  const getStreamEnrichmentState = useGetStreamEnrichmentState();
```

Comment:
Since `getStreamEnrichmentState` in only used for code flyout, does it make sense to have `useStreamEnrichmentCodeFlyout` instead which encapsulates the `isRequestPreviewFlyoutOpen`, `requestPreviewCodeContent`, `openRequestPreviewFlyout` and `closeRequestPreviewFlyout` and hides it from the component?

Or `useGetStreamEnrichmentState` exposes `{ ..., codeFlyout.*, ... }`?